File: client/types/resume.ts
export interface Resume {
  id: number;
  user_id: number;
  resume_text?: string;
  skills: string[];
  created_at?: string;
}

export interface Job {
  id: number;
  title: string;
  description: string;
  required_skills: string[];
  department?: string;
  created_at?: string;
}

export interface JobMatch {
  job: Job;
  match_score: number;
  skill_overlap_score: number;
  matching_skills: string[];
  missing_skills: string[];
}

export interface SkillMatch {
  skill: string;
  required: boolean;
  match: boolean;
  score: number;
  category: string;
}

export interface AnalysisResult {
  overallScore: number;
  skillMatches: SkillMatch[];
  strengths: string[];
  gaps: string[];
  recommendations: string[];
  experience: {
    required: string;
    candidate: string;
    match: boolean;
  };
  education: {
    required: string;
    candidate: string;
    match: boolean;
  };
}

File: schemas/resume.py
"""
Resume and job matching schemas
"""
from .typing import List, Optional
from .pydantic import BaseModel
from .datetime import datetime

class ResumeBase(BaseModel):
    """Base resume schema"""
    resume_text: str
    skills: List[str]

class ResumeCreate(ResumeBase):
    """Resume creation schema"""
    pass

class ResumeResponse(ResumeBase):
    """Resume response schema"""
    id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True

class JobBase(BaseModel):
    """Base job schema"""
    title: str
    description: str
    required_skills: List[str]
    department: Optional[str] = None

class JobCreate(JobBase):
    """Job creation schema"""
    pass

class JobResponse(JobBase):
    """Job response schema"""
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

class JobMatchBase(BaseModel):
    """Base job match schema"""
    match_score: float
    skill_overlap_score: float
    matching_skills: List[str]
    missing_skills: List[str]

class JobMatchResponse(JobMatchBase):
    """Job match response schema"""
    job: JobResponse

    class Config:
        from_attributes = True
